codegen/plugins/resolver/plugin.go

```diff
@@ -0,0 +1,62 @@
+package resolver
+
+import (
+	"context"
+
+	"github.com/99designs/gqlgen/codegen"
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/pkg/errors"
+	"github.com/vektah/gqlparser/ast"
+)
+
+type Plugin struct{}
+
+var plugin = &Plugin{}
+
+func (p *Plugin) Schema(cfg *codegen.Config) (*ast.Source, error) {
+	return &ast.Source{
+		Name:  "resolver plugin",
+		Input: "directive @resolver on FIELD_DEFINITION",
+	}, nil
+}
+
+func (p *Plugin) Execute(cfg *codegen.Config, schema *ast.Schema) error {
+	if cfg.Directives.ImplementationFor("resolver") != "" {
+		return errors.Errorf("directive implementation for resolver already exists")
+	}
+	cfg.Directives["resolver"] = codegen.DirectiveMapEntry{Implementation: "github.com/99designs/gqlgen/codegen/plugins/resolver.DirectiveNoop"}
+	for _, typ := range schema.Types {
+		for _, f := range typ.Fields {
+			if d := f.Directives.ForName("resolver"); d != nil {
+				if _, ok := cfg.Models[typ.Name]; !ok {
```

Comment:
This code probably reads a bit better leveraging zero values
```go
	modelCfg := cfg.Models[typ.Name] // is zero struct, not null

	// still required, maps can be nil
	if modelCfg.Fields == nil {
		modelCfg.Fields = make(map[string]codegen.TypeMapField)
	}
	fieldCfg := modelCfg.Fields[f.Name] // again, not ptr so zeroval is empty struct
	fieldCfg.ForceResolver = true
	modelCfg.Fields[f.Name] = fieldCfg
	cfg.Models[typ.Name] = modelCfg
```

Also much less likely for a new field in TypeMapEntry to be forgotten about here, causing this plugin to keep zeroing it out.

Reply:
As we discussed, the alternative is to have each plugin return a new config struct that is merged into the existing one.  I think this is the approach I'll take.